### ISSUE TITLE
Why this repo can not converge?

### ISSUE THREAD

**Issue by wangsssky** [eriklindernoren/PyTorch-YOLOv3#540]
Why this repo can not converge even after fixed the bugs, such as target index, padding error...? Can anyone give some hints? What can be the major difference between this and the Ultralytics repo?

**Comment by voodoopotato** [comment]
It's a difference in the objective. This repo's primary objective is to be able to run yolo model with pre-trained weights from darknet. Ultralytics aims to be a commercial product. The training code here is primarily a side effect. There was little effort putting in optimizing the Yolo model to the Pytorch since it only needs to be able to run pre-trained weights from DarkNet. If you take a peek at Ultralyitics's Resolved Issues dating back to 2018, you can see they had similar issues too, but have been resolved with various learning tricks.

**Comment by wangsssky** [comment]
> It's a difference in the objective. This repo's primary objective is to be able to run yolo model with pre-trained weights from darknet. Ultralytics aims to be a commercial product. The training code here is primarily a side effect. There was little effort putting in optimizing the Yolo model to the Pytorch since it only needs to be able to run pre-trained weights from DarkNet. If you take a peek at Ultralyitics's Resolved Issues dating back to 2018, you can see they had similar issues too, but have been resolved with various learning tricks.

Thank you for your reply. I took a glimpse at ultralytics/yolov3 tag7. I find that a lot of changes are applied to the yolo layer and build_target function after that point.

**Comment by jiu-jian-xian** [comment]
Do you haven't find the reason?

**Comment by wangsssky** [comment]
> Do you haven't find the reason?

Not yet.

**Comment by Flova** [comment]
The convergence on coco is now greatly improved.
